recaptcha/recaptcha-enterprise/src/main/kotlin/com/pingidentity/recaptcha/enterprise/ReCaptchaEnterpriseCallback.kt

```diff
@@ -0,0 +1,157 @@
+     * @param provider The custom RecaptchaClientProvider implementation
+     */
+    fun setProvider(provider: RecaptchaClientProvider) {
+        recaptchaClientProvider = provider
+    }
+}
+
+/**
+ * Journey callback for handling ReCaptcha Enterprise verification challenges.
+ *
+ * This callback integrates Google's ReCaptcha Enterprise service into Journey
+ * authentication flows. It handles the verification process by:
+ * - Receiving site key configuration from the server
+ * - Creating ReCaptcha clients using the configured provider
+ * - Executing verification actions with proper timeout handling
+ * - Returning verification tokens for server validation
+ *
+ * The callback supports customizable actions, timeouts, and client providers
+ * to accommodate different verification requirements and testing scenarios.
+ *
+ * Example usage:
+ * ```kotlin
+ * val callback = journey.next() as ReCaptchaEnterpriseCallback
+ * val result = callback.verify {
+ *     recaptchaAction = RecaptchaAction.LOGIN
+ *     timeoutInMills = 15000L
+ *     setProvider(myCustomProvider)
+ * }
+ * ```
+ */
+class ReCaptchaEnterpriseCallback : AbstractCallback(), JourneyAware {
+    /**
+     * The ReCaptcha site key received from the server configuration.
+     * This key is automatically populated during callback initialization.
+     */
+    var reCaptchaSiteKey: String = ""
+        private set
+
+    override lateinit var journey: Journey
+
+    /**
+     * Initializes callback properties from server configuration.
+     *
+     * @param name The property name from server configuration
+     * @param value The property value as JsonElement
+     */
+    override fun init(name: String, value: JsonElement) {
+        if ("recaptchaSiteKey" == name) {
+            this.reCaptchaSiteKey = value.jsonPrimitive.content
+        }
+    }
+
+    /**
+     * Internal method for testing purposes to initialize properties.
+     *
+     * @param name The property name
+     * @param value The property value as JsonElement
+     */
+    internal fun initForTest(name: String, value: JsonElement) {
+        init(name, value)
+    }
+
+    /**
+     * Performs ReCaptcha Enterprise verification with the specified configuration.
+     *
+     * This method orchestrates the entire verification process:
+     * 1. Applies the provided configuration block
+     * 2. Creates a ReCaptcha client using the configured provider
+     * 3. Executes the specified action with timeout
+     * 4. Returns the verification token for server validation
+     *
+     * @param block DSL configuration block for customizing verification parameters
+     * @return Result containing the verification token as JsonObject, or failure with exception
+     * @throws IllegalStateException if RecaptchaClientProvider is not configured
+     * @throws IllegalStateException if token verification fails
+     */
+    suspend fun verify(block: ReCaptchaEnterpriseConfig.() -> Unit = {}): Result<JsonObject> {
```

Comment:
The result should not be the Callback payload, it should just be the raw data which should be the reCaptcha result.
Since client.execute already return a Result, may consider this
```kotlin
 return kotlin.runCatching {
            val client =
                Recaptcha.fetchClient(
                    ContextProvider.context as Application,
                    config.reCaptchaSiteKey
                )
            val result = client.execute(config.recaptchaAction, config.timeoutInMills)
            val value = result.getOrThrow()
            super.input(value)
            value
        }
```